### PR TITLE
boards: adafruit: add initial support for feather m4 express

### PR DESCRIPTION
**Pull request by Terezvent**
This PR adds initial support for the [Adafruit feather m4 express](https://www.adafruit.com/product/3857) development board.

https://builds.zephyrproject.io/zephyr/pr/81081/docs/boards/adafruit/feather_m4_express/doc/index.html

### CODE REVIEW

File: boards/adafruit/feather_m4_express/adafruit_feather_m4_express.dts [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+	clock-frequency = <120000000>;
+};
+
+&sercom3 {
+	status = "okay";
+	compatible = "atmel,sam0-uart";
+	current-speed = <115200>;
+	rxpo = <1>;
+	txpo = <0>;
+
```
Comment (nordicjm) [inline boards/adafruit/feather_m4_express/adafruit_feather_m4_express.dts:49]:
```suggestion
```

File: boards/adafruit/feather_m4_express/adafruit_feather_m4_express.dts [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+	pinctrl-0 = <&sercom3_uart_default>;
+	pinctrl-names = "default";
+};
+
+&sercom1 {
+	status = "okay";
+	compatible = "atmel,sam0-spi";
+	dipo = <3>;
+	dopo = <0>;
+
```
Comment (nordicjm) [inline boards/adafruit/feather_m4_express/adafruit_feather_m4_express.dts:59]:
```suggestion
```

File: boards/adafruit/feather_m4_express/adafruit_feather_m4_express.dts [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+	#address-cells = <1>;
+	#size-cells = <0>;
+
```
Comment (nordicjm) [inline boards/adafruit/feather_m4_express/adafruit_feather_m4_express.dts:62]:
```suggestion
```

File: boards/adafruit/feather_m4_express/adafruit_feather_m4_express.dts [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+	pinctrl-0 = <&sercom1_spi_default>;
+	pinctrl-names = "default";
+};
+
+&tcc0 {
+	status = "okay";
+	compatible = "atmel,sam0-tcc-pwm";
+	prescaler = <8>;
+	#pwm-cells = <2>;
+
```
Comment (nordicjm) [inline boards/adafruit/feather_m4_express/adafruit_feather_m4_express.dts:72]:
```suggestion
```

File: boards/adafruit/feather_m4_express/adafruit_feather_m4_express.dts [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+	pinctrl-0 = <&pwm0_default>;
+	pinctrl-names = "default";
+};
+
+zephyr_udc0: &usb0 {
+	status = "okay";
+
```
Comment (nordicjm) [inline boards/adafruit/feather_m4_express/adafruit_feather_m4_express.dts:79]:
```suggestion
```

File: boards/adafruit/feather_m4_express/doc/index.rst [hunk 6]
```diff
@@ -0,0 +1,208 @@
...
+- Internal trimmed 8 MHz oscillator
+- A user LED
+- An RGB NeoPixel LED
+- Native USB port
+- One reset button
+
+Supported Features
+==================
+
+The adafruit_feather_m4_express board configuration supports the following
```
[the remark this answers is not in the document]
Comment (Terezvent) [reply boards/adafruit/feather_m4_express/doc/index.rst]: Done

File: boards/adafruit/feather_m4_express/doc/img/adafruit_feather_m4_express.jpg
[the remark this answers is not in the document]
Comment (nordicjm) [reply boards/adafruit/feather_m4_express/doc/img/adafruit_feather_m4_express.jpg:1]: does not seem to have been resolved

[the remark this answers is not in the document]
Comment (Terezvent) [reply boards/adafruit/feather_m4_express/doc/img/adafruit_feather_m4_express.jpg:1]: Sorry, now it's fixed

File: boards/adafruit/feather_m4_express/doc/index.rst [hunk 7]
```diff
@@ -0,0 +1,205 @@
...
+********
+
+The Adafruit Feather M4 Express is a compact, lightweight
+ARM development board with an onboard mini NeoPixel, 2 MiB
+of SPI flash, charging status indicator and user LEDs, USB
+connector, 21 GPIO pins and a small prototyping area.
+
+.. image:: img/adafruit_feather_m4_express.jpg
+     :align: center
+     :alt: Adafruit Feather M4 Express
```
Comment (kartben) [inline boards/adafruit/feather_m4_express/doc/index.rst:13]:
```suggestion
```

File: boards/adafruit/feather_m4_express/board.yml [hunk 8]
```diff
@@ -0,0 +1,5 @@
+board:
+  name: adafruit_feather_m4_express
```
Comment (kartben) [inline boards/adafruit/feather_m4_express/board.yml:2]:
```suggestion
  name: adafruit_feather_m4_express
  full_name: Feather M4 Express
```

File: boards/adafruit/feather_m4_express/doc/index.rst [hunk 9]
```diff
@@ -0,0 +1,205 @@
...
+more, such as the :zephyr:code-sample:`usb-cdc-acm` sample which sets up a virtual
+serial port that echos characters back to the host PC.
+
+Programming and Debugging
+*************************
+
+The Feather ships with a the BOSSA compatible UF2 bootloader.  The
+bootloader can be entered by quickly tapping the reset button twice.
+
+Additionally, if :code:`CONFIG_USB_CDC_ACM` is enabled then the bootloader
```
Comment (kartben) [inline boards/adafruit/feather_m4_express/doc/index.rst:115]:
```suggestion
Additionally, if :kconfig:option:`CONFIG_USB_CDC_ACM` is enabled then the bootloader
```

**Review by kartben — CHANGES_REQUESTED** [review]

File: boards/adafruit/feather_m4_express/doc/index.rst (hunk 7 above)
Comment (Terezvent) [reply boards/adafruit/feather_m4_express/doc/index.rst]: Fixed

File: boards/adafruit/feather_m4_express/doc/index.rst (hunk 9 above)
Comment (Terezvent) [reply boards/adafruit/feather_m4_express/doc/index.rst]: fixed

File: boards/adafruit/feather_m4_express/board.yml (hunk 8 above)
Comment (Terezvent) [reply boards/adafruit/feather_m4_express/board.yml:2]: fixed

**Review by nordicjm — APPROVED** [review]

File: boards/adafruit/feather_m4_express/doc/img/adafruit_feather_m4_express.webp
Comment (nordicjm) [inline boards/adafruit/feather_m4_express/doc/img/adafruit_feather_m4_express.webp:1]: Just noticed this is png still, can you convert to webp? E.g. using https://www.freeconvert.com/png-to-webp then put it through https://tinypng.com/

**Review by nordicjm — CHANGES_REQUESTED** [review]

**Review by nordicjm — APPROVED** [review]

**Comment by Terezvent** [comment]
Hello @kartben 
Could you help me review the changes you requested, please?

**Review by kartben — APPROVED** [review]
@Terezvent this is great, thanks!

**Pull request merged**